### ISSUE TITLE
Question about normalisation part

### ISSUE THREAD

**Issue by mortunco** [julian-gehring/SomaticSignatures#4]
Hi,

First of all thank you for creating a stable and useful software for mutation signature analysis. I spent more than a week dealing other softwares loading vcf file step. so thank you!

I would like to compare mutational signature of TF proximity mutations vs (genome - TF binding proximity regions). So I have to make a proper normalisation for the sequence context.

Can I solve my problem with ``` 4.8 Extension: Normalization of Sequence Motif Frequencies ```. I am thinking about giving my TF binding areas as a bed input instead of hg19 gene exons like you have explained in the introduction part. 

Any comment will be very useful.

Best regards,

Tunc.

**Comment by juliangehring** [comment]
I'm glad to hear that the package has been useful for your analysis!

Your description of your planned approach is very general, so it is a bit hard for me to give you a helpful comment. From what I understand, you would very much want to adjust for the different sequence composition when looking at only a small set of the genome. In this sense, following the analysis outlined in section 4.8 of the vignette seems the right way forward to me - as you already suggested. Without knowing the details of your data set, I would be somehow concerned that your target regions might leave you with too few variants to robustly infer mutational signatures (see #2 ).

**Comment by mortunco** [comment]
Dear Julian,

Thank you for answering very fast again! 

Before I get your permission, I have already followed this method. Let me rephrase myself; we are working on breast cancer snv data to characterise estrogen receptor proximity mutations. When I see the mutation context of ES proximity regions and non ES proximity region mutations, I found that one of the (T>C) is only enriched in ES proximity mutations. But to prove that it is independent from sequence context I need a normalisation. ( I have seen couple articles about TF proximity mutation signatures and they did a analysis to check if it is caused by sequence context , I need a method for either normalisation. [CTCF/cohesin-binding sites are frequently mutated in cancer](http://www.nature.com/ng/journal/v47/n7/full/ng.3335.html) ) 

So for the analysis, I normalised WGS sequence context to TF sequence context like you did for WES sequence context.  ( My script is below)

I would like to ask a question with inferring this normalisation. Hypothetically, by doing normalisation, I am normalising the WGS sequence context to TF sequence context ? Is this true ?

Thank you very much for your patience and help

```
#es250 is estrogen receptor proximity mutations.
ICGC_cohort=GRangesList(... = c(es250,`BRCA-US`,`BRCA-EU`,))

sca_motifs = mutationContext(c(unlist(ICGC_cohort)),ref = BSgenome.Hsapiens.1000genomes.hs37d5)
sample_motifsmatrix = motifMatrix(sca_motifs,group='sampleNames',normalize=T)

### NORMALIZATION ###
mybedfile = read.table('ER.bed',header = F,stringsAsFactors = F)
colnames(mybedfile)=c('chrom','start','stop','peakname','score')
mybedGrange=makeGRangesFromDataFrame(mybedfile,keep.extra.columns = T)

k=3

hs_chrs = as(seqinfo(BSgenome.Hsapiens.1000genomes.hs37d5), "GRanges")
hs_chrs = keepSeqlevels(hs_chrs, c(1:22, "X", "Y"))
k3_hs_chrs = kmerFrequency(BSgenome.Hsapiens.1000genomes.hs37d5, n = length(hs_TF), k, hs_chrs)

hs_TF = mybedGrange
hs_TF = ncbi(keepStandardChromosomes(hs_TF))
k3_TF = kmerFrequency(ref = BSgenome.Hsapiens.1000genomes.hs37d5,k = 3,n = length(hs_TF))

norms = k3_hs_chrs / k3_TF
#----------------------------------------------------------------------------------------#

sca_motifs_norm = normalizeMotifs(sample_motifsmatrix,norms)
```
Afterwards, I followed introduced method for finding signatures and plotting with ```sca_motifs_norm``` object.

**Comment by juliangehring** [comment]
The normalisation steps look reasonable and correct.

However, I'm rather sceptical that this will yield meaningful results in general due to the small number of variants in your data set (see #2), and normalising for the sequencing context cannot address this. I'd also keep in mind that for very small regions of interest, such as specific TF regions in your case, additional effects in the data generation may need accounting for.

**Comment by juliangehring** [comment]
An addition to my previous comment: I don't understand what `length(hs_TF)` represents from your code, and can't judge whether your code would give you reasonable kmer frequencies.

**Comment by mortunco** [comment]
I have have 4239 TF binding regions. Therefore, in order to incorparate  ```As an example, we compute the frequency of 3-mers for the human toplevel chromosomes, based on a sample of 10'000 locations.``` to my dataset, I thought putting number of peaks will work out. 

For the total number of mutations, I know that we dont have many mutations but I wonder can't we analyse it like a one big sample instead of identifying all patients one by one. Because mutation signature is not the main object of our study.
